fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java

```diff
@@ -465,10 +467,23 @@ public Boolean isNew() {
     public void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTranslator,
         final Model inputModel, final RdfStream originalTriples) throws MalformedRdfException {
 
+        LOGGER.info("Resource path: {}", getPath());
+
+        final StmtIterator stmtIterator = inputModel.listStatements();
+
+        while (stmtIterator.hasNext()) {
+            final Statement s = stmtIterator.next();
+            final Resource subject = s.getSubject();
+            if (!subject.toString().contains(getPath())) {
+                LOGGER.info("Input model subject not target of resource: {}", subject.toString());
+                throw new UnsupportedOperationException("Non targetted resource " + subject.toString());
```

Comment:
Let's subclass this to something more specific, a la `NonLocalTriplesException` or the like.